core/server/helm_release.go

```diff
@@ -0,0 +1,148 @@
+		return nil, err
+	}
+
+	var results []*pb.HelmRelease
+	for _, helmRelease := range l.Items {
+		results = append(results, types.HelmReleaseToProto(&helmRelease, []*pb.GroupVersionKind{}))
+	}
+
+	return &pb.ListHelmReleasesResponse{
+		HelmReleases: results,
+	}, nil
+}
+
+func (cs *coreServer) GetHelmRelease(ctx context.Context, msg *pb.GetHelmReleaseRequest) (*pb.GetHelmReleaseResponse, error) {
+	k8s, err := cs.k8s.Client(ctx)
+	if err != nil {
+		return nil, doClientError(err)
+	}
+
+	helmRelease := helmv2.HelmRelease{}
+
+	if err = get(ctx, k8s, msg.Name, msg.Namespace, &helmRelease); err != nil {
+		return nil, err
+	}
+
+	inventory, err := getHelmReleaseInventory(ctx, helmRelease, k8s)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetHelmReleaseResponse{
+		HelmRelease: types.HelmReleaseToProto(&helmRelease, inventory),
+	}, err
+}
+
+func getHelmReleaseInventory(ctx context.Context, helmRelease v2beta1.HelmRelease, k8s client.Client) ([]*pb.GroupVersionKind, error) {
+	storageNamespace := helmRelease.GetNamespace()
+	if helmRelease.Spec.StorageNamespace != "" {
+		storageNamespace = helmRelease.Spec.StorageNamespace
+	}
+
+	storageName := helmRelease.GetName()
+	if helmRelease.Spec.ReleaseName != "" {
+		storageName = helmRelease.Spec.ReleaseName
+	}
+
+	storageVersion := helmRelease.Status.LastReleaseRevision
+	if storageVersion < 1 {
+		// skip release if it failed to install
+		return nil, nil
+	}
+
+	storageSecret := v1.Secret{}
+	secretName := fmt.Sprintf("sh.helm.release.v1.%s.v%v", storageName, storageVersion)
+
+	if err := get(ctx, k8s, secretName, storageNamespace, &storageSecret); err != nil {
+		return nil, err
+	}
+
+	releaseData, releaseFound := storageSecret.Data["release"]
+	if !releaseFound {
+		return nil, fmt.Errorf("failed to decode the Helm storage object for HelmRelease '%s'", helmRelease.Name)
+	}
+
+	byteData, err := base64.StdEncoding.DecodeString(string(releaseData))
+	if err != nil {
+		return nil, err
+	}
+
+	var magicGzip = []byte{0x1f, 0x8b, 0x08}
+	if bytes.Equal(byteData[0:3], magicGzip) {
+		r, err := gzip.NewReader(bytes.NewReader(byteData))
+		if err != nil {
+			return nil, err
+		}
+
+		defer r.Close()
+
+		uncompressedByteData, err := io.ReadAll(r)
+		if err != nil {
+			return nil, err
+		}
+
+		byteData = uncompressedByteData
+	}
+
+	storage := types.HelmReleaseStorage{}
+	if err := json.Unmarshal(byteData, &storage); err != nil {
+		return nil, fmt.Errorf("failed to decode the Helm storage object for HelmRelease '%s': %w", helmRelease.Name, err)
+	}
+
+	objects, err := ssa.ReadObjects(strings.NewReader(storage.Manifest))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read the Helm storage object for HelmRelease '%s': %w", helmRelease.Name, err)
+	}
+
+	var gvk []*pb.GroupVersionKind
+
+	found := map[string]bool{}
+
+	for _, entry := range objects {
+		entry.GetAPIVersion()
+
+		idstr := strings.Join([]string{entry.GetAPIVersion(), entry.GetKind()}, "_")
```

Comment:
Do you know why we call `entry.GetAPIVersion` here and then again on 134? I can see we are doing it somewhere else in the code, so am curious what repeating but not keeping the value gives us.

Reply:
I have no idea, and because I have no idea, I just used the same logic XD

Reply:
does it hurt anything if you remove the first one?

Reply:
while testing it does not hurt, but I don't know if it does with real k8s objects coming from a real k8s cluster.

Reply:
ack, makes sense, would be good to prove and remove at somepoint

Reply:
Removed... Why?

```
entry.GetAPIVersion()

idstr := strings.Join([]string{entry.GetAPIVersion(), entry.GetKind()}, "_")
```

The function call sequence is:
1. `entry.GetAPIVersion()`
2. `entry.GetKind()`
3. `strings.Join()`

With that extra call the difference is an extra call only:
1. `entry.GetAPIVersion()`
2. `entry.GetAPIVersion()`
3. `entry.GetKind()`
4. `strings.Join()`

I see only one case when it can have any effects, if the underlying code is far from optimal and has an internal counter (how many times it was called) or something, and it does not make sense.